recipes/sharedtags-greedyview.lua

```diff
@@ -0,0 +1,195 @@
+        local seltags = newscreen.selected_tags
+
+        -- Move over all tags to an existing screen.
+        for _,tag in ipairs(s.tags) do
+            sharedtags.movetag(tag, newscreen)
+        end
+
+        -- Restore the viewed tags on the new screen.
+        for i,tag in ipairs(seltags) do
+            if i == 1 then
+                tag:view_only()
+            else
+                awful.tag.viewtoggle(tag)
+            end
+        end
+    end)
+end)
+
+--- Create new tag objects.
+-- The first tag defined for each screen will be automatically selected.
+-- @tparam table def A list of tables with the optional keys `name`, `layout`
+-- and `screen`. The `name` value is used to name the tag and defaults to the
+-- list index. The `layout` value sets the starting layout for the tag and
+-- defaults to the first layout. The `screen` value sets the starting screen
+-- for the tag and defaults to the first screen. The tags will be sorted in this
+-- order in the default taglist.
+-- @treturn table A list of all created tags. Tags are assigned numeric values
+-- corresponding to the input list, and all tags with non-numerical names are
+-- also assigned to a key with the same name.
+-- @usage local tags = sharedtags(
+--   -- "main" is the first tag starting on screen 2 with the tile layout.
+--   { name = "main", layout = awful.layout.suit.tile, screen = 2 },
+--   -- "www" is the second tag on screen 1 with the floating layout.
+--   { name = "www" },
+--   -- Third tag is named "3" on screen 1 with the floating layout.
+--   {})
+-- -- tags[2] and tags["www"] both refer to the same tag.
+function sharedtags.new(def)
+    local tags = {}
+
+    for i,t in ipairs(def) do
+        tags[i] = awful.tag.add(t.name or i, {
+            screen = (t.screen and t.screen <= capi.screen.count()) and t.screen or capi.screen.primary,
+            layout = t.layout,
+            sharedtagindex = i
+        })
+
+        -- Create an alias between the index and the name.
+        if t.name and type(t.name) ~= "number" then
+            tags[t.name] = tags[i]
+        end
+
+        -- If no tag is selected for this screen, then select this one.
+        if not tags[i].screen.selected_tag then
+            tags[i]:view_only() -- Updates the history as well.
+        end
+    end
+
+    return tags
+end
+
+--- Move the specified tag to a new screen, if necessary.
+-- @param tag The tag to move.
+-- @tparam[opt=awful.screen.focused()] number screen The screen to move the tag to.
+-- @treturn bool Whether the tag was moved.
+function sharedtags.movetag(tag, screen)
+    screen = screen or awful.screen.focused()
+    local oldscreen = tag.screen
+
+    -- If the specified tag is allocated to another screen, we need to move it.
+    if oldscreen ~= screen then
+        -- greedyview xmonad-style
+        if tag.selected then
+            local newsel = screen.selected_tag
+            newsel.screen = oldscreen
```

Comment:
Why do this in two lines instead of just `screen.selected_tag.screen = oldscreen`?